client/extensions/woocommerce/app/settings/payments/stripe/payment-method-stripe-connected-dialog.js

```diff
@@ -0,0 +1,164 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { localize } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import AuthCaptureToggle from 'woocommerce/components/auth-capture-toggle';
+import Dialog from 'components/dialog';
+import FormFieldset from 'components/forms/form-fieldset';
+import FormLabel from 'components/forms/form-label';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
+import FormTextInput from 'components/forms/form-text-input';
+import PaymentMethodEditFormToggle from '../payment-method-edit-form-toggle';
+import StripeConnectAccount from './payment-method-stripe-connect-account';
+
+class PaymentMethodStripeConnectedDialog extends Component {
+
+	static propTypes = {
+		stripeConnectAccount: PropTypes.shape( {
+			connectedUserID: PropTypes.string,
+			displayName: PropTypes.string,
+			email: PropTypes.string,
+			firstName: PropTypes.string,
+			isActivated: PropTypes.bool,
+			lastName: PropTypes.string,
+			logo: PropTypes.string,
+		} ),
+		method: PropTypes.shape( {
+			settings: PropTypes.shape( {
+				apple_pay: PropTypes.shape( { value: PropTypes.string.isRequired } ).isRequired,
+				capture: PropTypes.shape( { value: PropTypes.string.isRequired } ).isRequired,
+				secret_key: PropTypes.shape( { value: PropTypes.string.isRequired } ).isRequired,
+				publishable_key: PropTypes.shape( { value: PropTypes.string.isRequired } ).isRequired,
+				testmode: PropTypes.shape( { value: PropTypes.string.isRequired } ).isRequired,
+				test_publishable_key: PropTypes.shape( { value: PropTypes.string.isRequired } ).isRequired,
+				test_secret_key: PropTypes.shape( { value: PropTypes.string.isRequired } ).isRequired,
+			} ),
+		} ),
+		onCancel: PropTypes.func.isRequired,
+		onEditField: PropTypes.func.isRequired,
+		onDone: PropTypes.func.isRequired,
+		site: PropTypes.shape( {
+			domain: PropTypes.string.isRequired,
+		} ),
+	};
+
+	////////////////////////////////////////////////////////////////////////////
+	// Live vs Test Mode and API Key Fields
+
+	onSelectLive = () => {
+		this.props.onEditField( { target: { name: 'testmode', value: 'no' } } );
+	}
+
+	onSelectTest = () => {
+		this.props.onEditField( { target: { name: 'testmode', value: 'no' } } );
+	}
+
+	onSelectAuthOnly = () => {
+		this.props.onEditField( { target: { name: 'capture', value: 'no' } } );
+	}
+
+	onSelectCapture = () => {
+		this.props.onEditField( { target: { name: 'capture', value: 'yes' } } );
+	}
+
+	getStatementDescriptorPlaceholder = () => {
+		const { site, translate } = this.props;
+		const domain = site.domain.substr( 0, 22 ).trim().toUpperCase();
```

Comment:
This logic also appears in `PaymentMethodStripeKeyBasedDialog` - might be a good candidate to extract to a selector or a util function.

Reply:
Fixed in https://github.com/Automattic/wp-calypso/pull/17197/commits/4b5f156ddc505a0e4a4f4271a7531f008651485e